### PR TITLE
Store tokenizer conversion params in rt_info / refactor passing params

### PR DESCRIPTION
**Pull request by pavel-esir**
CVS-146675

### CODE REVIEW

File: python/openvino_tokenizers/utils.py [hunk 1]
```diff
@@ -10,15 +10,66 @@
 from openvino import Model, Type
 from openvino.preprocess import PrePostProcessor
 from openvino.runtime import opset12 as opset
+from dataclasses import dataclass, field, asdict
+from transformers import PreTrainedTokenizerBase
```
Comment (apaniukov) [inline python/openvino_tokenizers/utils.py:14]: I would avoid global import from transformers since it is an optional dependency.

Comment (pavel-esir) [reply python/openvino_tokenizers/utils.py]: done

File: python/openvino_tokenizers/convert_tokenizer.py [hunk 2]
```diff
@@ -10,14 +10,47 @@
...
+                          "convert_tokenizer(tokenizer_object, with_detokenizr=True, add_special_tokens=True,...). " \
+                          "Or within 'params' argument, e.g. " \
+                          "convert_tokenzier(tokenizer_object, params={'with_detokenizr': True, 'add_special_tokens': True, ...})"
+                    raise ValueError(msg)
+        
+        return func(*argc, **kwargs)
+    
+    # Embed convert_tokenizer docstring with TokenzierConversionParams docstring.
+    pos = func.__doc__.find('    Returns:')
+    wrapper.__doc__ = func.__doc__[:pos] + TokenzierConversionParams.__doc__ + '\n' + func.__doc__[pos:]
```
Comment (apaniukov) [inline python/openvino_tokenizers/convert_tokenizer.py:45]: https://docs.python.org/3/library/functools.html#functools.wraps

File: python/openvino_tokenizers/convert_tokenizer.py [hunk 3]
```diff
@@ -10,14 +10,47 @@
...
+            for key in TokenzierConversionParams.__match_args__:
+                if kwargs[key] is not None:
+                    msg = "Cannot specify both 'params' and individual convert_tokenizer arguments simultaneously. " \
+                          "Please pass all conversion params either individually, e.g. " \
+                          "convert_tokenizer(tokenizer_object, with_detokenizr=True, add_special_tokens=True,...). " \
+                          "Or within 'params' argument, e.g. " \
+                          "convert_tokenzier(tokenizer_object, params={'with_detokenizr': True, 'add_special_tokens': True, ...})"
+                    raise ValueError(msg)
+        
+        return func(*argc, **kwargs)
```
Comment (apaniukov) [inline python/openvino_tokenizers/convert_tokenizer.py:41]:
```suggestion
    def wrapper(*args, **kwargs):
        params = None
        if len(args) > 1 and args[1] != None:
            params = args[1]
        if 'params' in kwargs:
            params = kwargs['params']
        
        if params is not None:
            for key in TokenzierConversionParams.__match_args__:
                if kwargs[key] is not None:
                    msg = "Cannot specify both 'params' and individual convert_tokenizer arguments simultaneously. " \
                          "Please pass all conversion params either individually, e.g. " \
                          "convert_tokenizer(tokenizer_object, with_detokenizr=True, add_special_tokens=True,...). " \
                          "Or within 'params' argument, e.g. " \
                          "convert_tokenzier(tokenizer_object, params={'with_detokenizr': True, 'add_special_tokens': True, ...})"
                    raise ValueError(msg)
        
        return func(*args, **kwargs)
```

File: python/openvino_tokenizers/convert_tokenizer.py (hunk 2 above)
Comment (pavel-esir) [reply python/openvino_tokenizers/convert_tokenizer.py]: Applied here @wraps and other stuff we discussed on teams.

File: python/openvino_tokenizers/convert_tokenizer.py [hunk 4]
```diff
@@ -10,14 +10,55 @@
...
+    change_outputs_type,
+    update_rt_info,
+    TokenzierConversionParams,
+)
 
 logger = logging.getLogger(__name__)
 
 
+def capture_arg(func):
+    from functools import wraps
```
Comment (apaniukov) [inline python/openvino_tokenizers/convert_tokenizer.py:24]: Move imports from the standard library to the top.

File: tests/tokenizers_test.py [hunk 5]
```diff
@@ -894,3 +894,54 @@ def test_rt_info_sentencepiece(hf_sentencepiece_tokenizers, is_sentencepiece_bac
...
+]
+@pytest.fixture(scope="session", params=models_to_check_rt_info)
+def tokenizer_to_check_rt_info(request):
+    return get_hf_tokenizer(request, trust_remote_code=True)
+
+
+def test_rt_info_conversion_params(tokenizer_to_check_rt_info):
+    import openvino as ov
+    from openvino_tokenizers import convert_tokenizer
+    from openvino_tokenizers.utils import TokenzierConversionParams
```
Comment (apaniukov) [inline tests/tokenizers_test.py:913]: These imports can be moved to the top.

**Review by apaniukov — APPROVED** [review]

File: tests/tokenizers_test.py (hunk 5 above)
Comment (pavel-esir) [reply tests/tokenizers_test.py]: done

File: python/openvino_tokenizers/convert_tokenizer.py (hunk 4 above)
Comment (pavel-esir) [reply python/openvino_tokenizers/convert_tokenizer.py]: done

**Pull request merged**